localstack-core/localstack/services/apigateway/next_gen/execute_api/parameters_mapping.py

```diff
@@ -0,0 +1,160 @@
+    path: dict[str, str]
+    querystring: dict[str, str | list[str]]
+
+
+class ParametersMapper:
+    def map_integration_request(
+        self,
+        request_parameters: dict[str, str],
+        invocation_request: InvocationRequest,
+        context_variables: ContextVariables,
+        stage_variables: dict[str, str],
+    ) -> RequestDataMapping:
+        request_data_mapping = RequestDataMapping(
+            header={},
+            path={},
+            querystring={},
+        )
+        # TODO: maybe extract functionality and re-use in `map_integration_response`
+
+        for integration_mapping, request_mapping in request_parameters.items():
+            integration_param_location, param_name = integration_mapping.removeprefix(
+                "integration.request."
+            ).split(".")
+
+            if request_mapping.startswith("method.request."):
+                method_req_expr = request_mapping.removeprefix("method.request.")
+                value = self._retrieve_parameter_from_invocation_request(
+                    method_req_expr, invocation_request
+                )
+
+            elif request_mapping.startswith("context."):
+                context_var_expr = request_mapping.removeprefix("context.")
+                value = self._retrieve_parameter_from_context_variables(
+                    context_var_expr, context_variables
+                )
+
+            elif request_mapping.startswith("stageVariables."):
+                stage_var_name = request_mapping.removeprefix("stageVariables.")
+                value = self._retrieve_parameter_from_stage_variables(
+                    stage_var_name, stage_variables
+                )
+
+            elif request_mapping.startswith("'") and request_mapping.endswith("'"):
+                value = request_mapping.strip("'")
+
+            else:
+                # TODO: verify this
+                value = None
+
+            if value:
+                request_data_mapping[integration_param_location][param_name] = value
+
+        return request_data_mapping
+
+    def map_integration_response(self):
+        pass
+
+    def _retrieve_parameter_from_invocation_request(
+        self, expr: str, invocation_request: InvocationRequest
+    ) -> str | list[str]:
+        if expr.startswith("body"):
+            body = invocation_request["body"] or b"{}"
+            body = body.strip()
+            # AWS only tries to JSON decode the body if it starts with some leading characters ({, [, ", ')
+            # otherwise, it ignores it
+            try:
+                decoded_body = self._json_load(body)
+            except ValueError:
+                raise Default4xxError(message="Invalid JSON in request body")
+
+            if expr == "body":
+                return to_str(body)
+
+            elif expr.startswith("body."):
+                json_path = expr.removeprefix("body.")
+                return self._get_json_path_from_dict(decoded_body, json_path)
```

Comment:
nit: In case there is a malformed expression, we should probably make sure that we don't continue matching. Maybe we should add a else and LOG something and return `None`?